File: Daily Challanges/2026-02-04.py
def truncate_text(text):
    if len(text) < 21:
        return text
    return text[:17] + "..."

test1 = truncate_text("Hello, world!")
test2 = truncate_text("This string should get truncated.")
test3 = truncate_text("Exactly twenty chars")
test4 = truncate_text(".....................")

check1 = "Hello, world!"
check2 = "This string shoul..."
check3 = "Exactly twenty chars"
check4 = "...................."

print(f"Test1: {test1 == check1}")
print(f"Test2: {test2 == check2}")
print(f"Test3: {test3 == check3}")
print(f"Test4: {test4 == check4}")

File: Daily Challanges/2026-02-04.js
function truncateText(text) {
  return text.length <= 20 ? text : text.slice(0, 17) + "...";
}

const test1 = truncateText("Hello, world!");
const test2 = truncateText("This string should get truncated.");
const test3 = truncateText("Exactly twenty chars");
const test4 = truncateText(".....................");

const check1 = "Hello, world!";
const check2 = "This string shoul...";
const check3 = "Exactly twenty chars";
const check4 = "....................";

console.log(`Test1: ${test1 === check1}`);
console.log(`Test2: ${test2 === check2}`);
console.log(`Test3: ${test3 === check3}`);
console.log(`Test4: ${test4 === check4}`);
